lib/iris/cube.py

```diff
@@ -755,6 +751,10 @@ def __init__(self, data, standard_name=None, long_name=None,
         # Cell Measures
         self._cell_measures_and_dims = []
 
+        self.fill_value = fill_value
+
+        self._dtype = dtype
```

Comment:
Assigning args and kwargs into the instance in the constructor should be done in the order the args and kwargs are passed to the constructor, as is true for all the other args.